syncer/ddl.go

```diff
@@ -138,6 +154,29 @@ func (s *Syncer) splitAndFilterDDL(
 
 		// get real tableNames before apply block-allow list
 		if s.onlineDDL != nil {
+			if onlineDDLFinish && len(tables) > 1 {
```

Comment:
I suggest move this local into online ddl package with a parameter of `[]*ddlInfo`. We can still first parseStatement and FetchTableNames, but then wrap the result into a `[]ddlInfo`. And the do the online ddl logic and finally other remaining logics.